packages/connector-facebook/src/index.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * Reference: Manually Build a Login Flow
+ * https://developers.facebook.com/docs/facebook-login/manually-build-a-login-flow
+ */
+
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  ConnectorMetadata,
+  GetAccessToken,
+  GetAuthorizationUri,
+  GetUserInfo,
+  ValidateConfig,
+  SocialConnector,
+  GetConnectorConfig,
+} from '@logto/connector-types';
+import { assert } from '@silverhand/essentials';
+import got, { RequestError as GotRequestError } from 'got';
+import { stringify } from 'query-string';
+
+import {
+  accessTokenEndpoint,
+  authorizationEndpoint,
+  scope,
+  userInfoEndpoint,
+  defaultMetadata,
+  defaultTimeout,
+  facebookConfigGuard,
+  FacebookConfig,
+} from './constant';
+
+export class FacebookConnector implements SocialConnector {
+  public metadata: ConnectorMetadata = defaultMetadata;
+
+  public readonly getConfig: GetConnectorConfig<FacebookConfig>;
+
+  constructor(getConnectorConfig: GetConnectorConfig<FacebookConfig>) {
+    this.getConfig = getConnectorConfig;
+  }
+
+  public validateConfig: ValidateConfig = async (config: unknown) => {
+    const result = facebookConfigGuard.safeParse(config);
+
+    if (!result.success) {
+      throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, result.error.message);
+    }
+  };
+
+  public getAuthorizationUri: GetAuthorizationUri = async (redirectUri, state) => {
+    const config = await this.getConfig(this.metadata.id);
+
+    return `${authorizationEndpoint}?${stringify({
+      client_id: config.clientId,
+      redirect_uri: redirectUri,
+      response_type: 'code',
+      state,
+      scope, // Only support fixed scope for v1.
+    })}`;
```

Comment:
seems that it's unnecessary to use `query-string.stringify` and `query-string`.

e.g.
![image](https://user-images.githubusercontent.com/10594507/165446164-91597dd3-d48d-4e35-803e-28bd1025b85e.png)

You can refactor it in another PR.